packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/FormStep/useFormStep.ts

```diff
@@ -0,0 +1,143 @@
+import { useCallback, useContext } from "react";
+import {
+    DropDestination,
+    DropSource,
+    DropTarget,
+    FbFormModelField,
+    FbFormStep,
+    DropPosition
+} from "~/types";
+import { useFormEditor } from "~/admin/components/FormEditor";
+import { DragObjectWithFieldInfo } from "~/admin/components/FormEditor/Droppable";
+import cloneDeep from "lodash/cloneDeep";
+import { FormStepContext } from "./FormStepContext/FormStepContext";
+
+interface CreateCustomFieldParams {
+    data: FbFormModelField;
+    dropDestination: DropDestination;
+}
+
+interface HandleDropParams {
+    item: DragObjectWithFieldInfo;
+    destinationPosition: DropPosition;
+    formStep: FbFormStep;
+}
+
+export const useFormStep = () => {
+    const { data, moveRow, moveField, getFieldPlugin, insertField } = useFormEditor();
+
+    const { editingField, dropDestination, setEditingField, setDropDestination } =
+        useContext(FormStepContext);
+
+    const editField = useCallback((field: FbFormModelField | null) => {
+        if (!field) {
+            setEditingField(null);
+            return;
+        }
+        setEditingField(cloneDeep(field));
+    }, []);
+
+    const onFormStepDrop = useCallback(
+        (params: HandleDropParams) => {
```

Comment:
```suggestion
        (params: OnFormStepDropParams) => {
```

Also, does it make sense to to maybe just undo the name change here to `handleDrop` since we don't have the original one anymore?